kusto/test/etoe/etoe_test.go

```diff
@@ -581,6 +580,161 @@ func TestQueries(t *testing.T) {
 	}
 }
 
+func TestStatment(t *testing.T) {
+	t.Parallel()
+
+	if skipETOE || testing.Short() {
+		t.Skipf("end to end tests disabled: missing config.json file in etoe directory")
+	}
+
+	_, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	client, err := kusto.New(testConfig.kcsb)
+	if err != nil {
+		panic(err)
+	}
+
+	t.Cleanup(func() {
+		t.Log("Closing client")
+		require.NoError(t, client.Close())
+		t.Log("Closed client")
+	})
+
+	allDataTypesTable := fmt.Sprintf("goe2e_all_data_types_%d_%d", time.Now().UnixNano(), rand.Int())
+	require.NoError(t, createIngestionTable(t, client, allDataTypesTable, true))
+
+	tests := []struct {
+		// desc is a description of a test.
+		desc string
+		// stmt is the Kusot Stmt that will be sent.
+		stmt kusto.Statement
+		// setup is a function that will be called before the test runs.
+		setup func() error
+		// teardown is a functiont that will be called before the test ends.
+		teardown func() error
+		qcall    queryFunc
+		mcall    mgmtFunc
+		qjcall   queryJsonFunc
+		options  interface{} // either []kusto.QueryOption or []kusto.MgmtOption
+		// doer is called from within the function passed to RowIterator.Do(). It allows us to collect the data we receive.
+		doer func(row *table.Row, update interface{}) error
+		// gotInit creates the variable that will be used by doer's update argument.
+		gotInit func() interface{}
+		// want is the data we want to receive from the query.
+		want interface{}
+		// should the test fail
+		failFlag bool
+	}{
+		{
+			desc: "New query method",
+			stmt: kql.NewStatementBuilder("table(tableName) | where vstr == txt"),
+			options: []kusto.QueryOption{kusto.QueryParameters(*kql.NewStatementQueryParameters().
+				AddString("tableName", "string", allDataTypesTable).
+				AddString("txt", "string", "asdf"))},
+			qcall: client.Query,
+			doer: func(row *table.Row, update interface{}) error {
+				if row.Replace {
+					fmt.Println("---") // Replace flag indicates that the query result should be cleared and replaced with this row
```

Comment:
Do a test with all of the data types, making sure they all pass correctly and have the correct results.